app/main.py

```diff
@@ -1 +1,37 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        if self not in Animal.alive:
+            Animal.alive.append(self)
```

Comment:
It's impossible that self is in Animal.alive, thus this condition is redundant. When you create a new object, how it can be in this list? Like this condition makes sense when you check if an animal of this type or with this name is in that list, but when you just use "in" it's impossible because python uses object identifier to check it